airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,461 @@
+                # If response list is not empty, append it to the running list.
+                name_collection += filter(None, response.get(response_key))
+                token = response.get("nextToken")
+
+            self.log.info("Retrieved list of %s %s.", len(name_collection), response_key)
+            if verbose:
+                self.log.info("%s found: %s", response_key.title(), name_collection)
+
+        except ClientError as e:
+            self.log.error(e.response["Error"]["Message"])
+            raise e
+        return name_collection
+
+
+def generate_config_file(
+    eks_cluster_name: str,
+    eks_namespace_name: str,
+    pod_name: str,
+    aws_profile: Optional[str],
+    pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+    pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+    aws_region: Optional[str] = None,
+) -> str:
+    """
+    Writes the kubeconfig file given an EKS Cluster.
+
+    :param eks_cluster_name: The name of the cluster to create the EKS Managed Nodegroup in.
+    :type eks_cluster_name: str
+    :param eks_namespace_name: The namespace to run within kubernetes.
+    :type eks_namespace_name: str
+    :param pod_name: The unique name to give the pod.  Used as an identifier in the config filename.
+    :type pod_name: str
+    :param aws_profile: The named profile containing the credentials to use.
+    :type aws_profile: str
+    :param pod_username: The username under which to execute the pod.
+    :type pod_username: str
+    :param pod_context: The name of the context access parameters to use.
+    :type pod_context: str
+    :param aws_region: The name of the AWS Region the EKS Cluster resides in.
+    :type aws_region: str
+    """
+    # Set up the client
+    session = boto3.Session(region_name=aws_region, profile_name=aws_profile)
+    eks_client = session.client("eks")
+
+    # Get cluster details
+    cluster = eks_client.describe_cluster(name=eks_cluster_name)
+    cluster_cert = cluster["cluster"]["certificateAuthority"]["data"]
+    cluster_ep = cluster["cluster"]["endpoint"]
+
+    token = _get_bearer_token(session=session, cluster_id=eks_cluster_name, aws_region=aws_region)
+
+    cluster_config = {
+        "apiVersion": "v1",
+        "kind": "Config",
+        "clusters": [
+            {
+                "cluster": {"server": cluster_ep, "certificate-authority-data": cluster_cert},
+                "name": eks_cluster_name,
+            }
+        ],
+        "contexts": [
+            {
+                "context": {
+                    "cluster": eks_cluster_name,
+                    "namespace": eks_namespace_name,
+                    "user": pod_username,
+                },
+                "name": pod_context,
+            }
+        ],
+        "current-context": pod_context,
+        "preferences": {},
+        "users": [
+            {
+                "name": pod_username,
+                "user": {
+                    "token": token,
```

Comment:
This token is only valid for an hour, but this operator can be used to run a longer job. We should take care of its rotation of this key to ensure it's valid.

Unfortunately, it won't be trivial as we have to write a new [client-go-credential-plugin](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins). To do this, you should write a new CLI tool that will do the generate a new credential.

Some tips on adding a new CLI tool to providers:
- You cannot modify the Airflow core so that this operator can be used on different Airflow versions and not create a dependency between the vendor and the core. You cannot modify any file in `airflow.cli` package.
- You can create a new [executable module](https://docs.python.org/3/tutorial/modules.html#executing-modules-as-scripts). To run this module, it is best to use python module resolution i.e. you should use [python -m](https://docs.python.org/3/using/cmdline.html#cmdoption-m). BTW, To run airflow, you can use `airflow` or `python -m airflow`. See: https://github.com/apache/airflow/pull/7808
- The `python` program may not refer to the current interpreter. You should [`sys.executable`](https://docs.python.org/3/library/sys.html#sys.executable) instead.

For example, see: [`airflow.providers.google.common.utils.id_token_credentials`](https://github.com/apache/airflow/blob/866a601b76e219b3c043e1dbbc8fb22300866351/airflow/providers/google/common/utils/id_token_credentials.py#L24])

Reply:
For more info about kubectl exec auth, see: 
https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins